core/twip/generic_task.py

```diff
@@ -0,0 +1,216 @@
+
+    def load_config(
+        self,
+        headless: bool,
+        device: str,
+        num_envs: int,
+        config: Dict[Any, str] = {},
+    ) -> None:
+        config["device"] = device
+        config["headless"] = headless
+
+        config["num_envs"] = num_envs
+        config["max_episode_length"] = 256
+
+        config["num_agents"] = 1
+        config["num_observations"] = 4
+        config["num_actions"] = 2
+        config["num_states"] = 0
+
+        config["observation_space"] = gym.spaces.Box(
+            low=np.array(
+                [
+                    -np.pi,
+                    -np.Inf,
+                    -1.0,
+                    -1.0,
+                ]
+            ),
+            high=np.array(
+                [
+                    np.pi,
+                    np.Inf,
+                    1.0,
+                    1.0,
+                ]
+            ),
+        )
+        config["action_space"] = gym.spaces.Box(
+            np.ones(config["num_actions"]) * -1.0,
+            np.ones(config["num_actions"]) * 1.0,
+        )
+        config["state_space"] = gym.spaces.Box(
+            np.ones(config["num_states"]) * -np.Inf,
+            np.ones(config["num_states"]) * np.Inf,
+        )
+
+        # task-specific config
+        config["domain_randomization"] = {}
+
+        print(f"{self.__class__.__name__} loaded config {config}")
+
+        super().load_config(
+            headless=headless,
+            device=device,
+            num_envs=num_envs,
+            config=config,
+        )
+
+    def construct(self) -> bool:
+        self.env = self.env_factory()
+        self.agent = self.agent_factory()
+
+        self.env.construct(self.agent)
+
+        return True
+
+    # RL-Games methods (required from IVecEnv)
+    def step(
+        self, actions: torch.Tensor
+    ) -> Tuple[
+        Dict[str, torch.Tensor], torch.Tensor, torch.Tensor, Dict[str, Any] | None
+    ]:
+        # goes through a RL step (includes all the base RL things such as get obs, apply actions, etc.
+        # args: actions to apply to the env
+        # returns: obs, rewards, resets, info
+
+        if actions is None:
+            return (
+                {"obs": self.obs_buf},
+                self.rewards_buf,
+                self.dones_buf,
+                None,
+            )
+
+        env_info = self.get_env_info()
+        self.progress_buf += 1
+
+        # shape of twip_imu_obs: (num_envs, 10)
+        # (:, 0:3) -> linear acceleration
+        # (:, 3:6) -> angular velocity
+        # (:, 6:10) -> quaternion (WXYZ)
+        twip_imu_obs = self.env.step(actions, render=not self.headless).to(
+            device=self.device
+        )  # is on GPU, so all subsequent calculations will be on GPU
+
+        # get the roll angle only
+        twip_roll = roll_from_quat(twip_imu_obs[:, 6:10])
+
+        self.obs_buf[:, 0] = twip_roll
+        self.obs_buf[:, 1] = twip_imu_obs[:, 5]  # angular velocity on z-axis
```

Comment:
Is this gravity free acceleration or gravity taken into account? 
To clarify, when the sensor is stationary, the imu will typically read [0, 0, 1g].
Gravity free acceleration doesn't account for, well, gravity he imu will typically read [0, 0, 0]when stationary. 
Need to know to know which values should be fed into the model